### PR TITLE
Enhancement/120 implementing a favourites page

### PR DESCRIPTION
**Pull request by rishab-goswami**
GitHub Issue (If applicable): [#120](https://github.com/softeng-701-group-5/softeng-701-assignment-1/issues/120)

## PR Type

What kind of change does this PR introduce?
Feature

## What is the current behavior?
No favourites page (right now it has the same functionality as the feed page). Also, the feedr title on the top of the feed page doesn't do anything. 

## What is the new behavior?
There is a new favourites page. Also, 2 new buttons have been added. "feedr" which takes the user back to the feed page and "favourites" which takes the user to the favourites page.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] Tested code with current upstream repo
- [x] Added new tests to the test suite for the feature
- [x] Tested the full test suite, and ensured that the feature does not break current build.
- [ ] Docs have been added/updated which fit (for bug fixes / features)
- [x] Contains **NO** breaking changes
- [x] Associated with an issue (GitHub or internal)
- [x] Checked licensing of Frameworks/Libraries (if applicable) 

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: client/src/pages/FavouritesPage.jsx [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+
+      <Container className={classes.container}>
+        <Grid container spacing={3} justify="center">
+          {loader && (
+            <CircularProgress className={classes.loader}></CircularProgress>
+          )}
+          {feed.map(
+            (item, i) =>
+              filters.includes(item.media) &&
+              // Only checking the mainText if there is text to check, untherwise it will come up as 'undefined'
```
Comment (lakshyaoberoi) [inline client/src/pages/FavouritesPage.jsx:118]: *otherwise (yes I know it is only just a comment)

**Comment by lakshyaoberoi** [comment]
Good to see you making some test cases as well! 👍

File: client/src/pages/FavouritesPage.jsx [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+              color="inherit"
+            >
+              FAVOURITES
+            </Button>
+            <SearchBox setSearch={setSearch} />
+          </Toolbar>
+        </Container>
+      </AppBar>
+
+      {!loader && <FilterBar setFilters={setFilters} />}
```
Comment (baja156) [inline client/src/pages/FavouritesPage.jsx:108]: I think the `<AppBar/>` and the filterBar can be reusable components but we should do it in a different issue for it

File: client/src/pages/FavouritesPage.jsx [hunk 3]
```diff
@@ -0,0 +1,134 @@
+import React from 'react';
+import {
+  AppBar,
+  //Typography,
```
Comment (baja156) [inline client/src/pages/FavouritesPage.jsx:4]: Remove pls :)

**Pull request merged**